Repo/Repo.cs

```diff
@@ -9,9 +9,9 @@ public interface IRepository<Todo, TIndex>
      {
         Todo CreateTodo(Todo request);
         Todo UpdateTodo(TIndex id, Todo request);
-        Dictionary<Guid, Todo> DeleteTodo(TIndex id);
+        List<Todo> DeleteTodo(TIndex id);
         Todo GetATodo(TIndex id);
-        Dictionary<Guid, Todo>  GetTodos();
+        List<Todo> GetTodos();
```

Comment:
You are typing the return as a `List<T>`. As a rule you should try to use only the most general type possible to achieve the result.

You can determine what type your return should be based on the assumed needs of the client class. In this case, the only need communicated is a need to be able to enumerate the elements in the list. (Able to use a for-loop on the collection).

`IEnumerable<T>` is the more appropriate type as it only exposes the enumerator to the calling code.